src/core/compact_object_test.cc

```diff
@@ -34,6 +34,36 @@ void PrintTo(const CompactObj& cobj, std::ostream* os) {
   *os << "cobj: [" << cobj.ObjType() << "]";
 }
 
+// This is for the mimalloc test - being able to find an address in memory
+// where we have memory underutilzation
+// see issue number 448 (https://github.com/dragonflydb/dragonfly/issues/448)
+std::vector<void*> AllocateForTest(int size, std::size_t allocate_size, int factor1 = 1,
+                                   int factor2 = 1) {
+  std::vector<void*> ptrs;
+  for (int index = 0; index < size; index++) {
+    auto alloc_size = index % 13 == 0 ? allocate_size * factor1 : allocate_size * factor2;
+    auto heap_alloc = mi_heap_get_backing();
+    void* ptr = mi_heap_malloc(heap_alloc, alloc_size);
+    ptrs.push_back(ptr);
+  }
+  return ptrs;
+}
+
+bool FindUnderutilizedMemory(const std::vector<void*>& ptrs, float ratio) {
+  auto it = std::find_if(ptrs.begin(), ptrs.end(), [&](auto p) {
+    int r = p && zmalloc_page_is_underutilized(p, ratio);
+    return r > 0;
+  });
+  return it != ptrs.end();
+}
+
+void DecimateMemory(std::vector<void*>* ptrs, size_t steps) {
```

Comment:
i am not sure I understand the meaning of Decimate here. What you are trying to say?

Reply:
I will change the name of the function to something with better naming